.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,46 @@
+name: Doxygen
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  Build and Publish Documentation:
+    name: Build and Publish Documentation
+    runs-on: ubuntu-latest
+    container:
+      image: novelrt/novelrt-build:latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Create documentation branch
+        run: git checkout -b documentation/$GITHUB_SHA
+
+      #Mandatory step it appears otherwise pushing the commit will fail
+      #Username and Email found from https://github.community/t/github-actions-bot-email-address/17204/3
+      - name: Configure Git
+        run: |
+          git config user.name "github-actions"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+      - name: Generate Documentation
+        run: scripts/cibuild.sh --documentation -DDOXYGEN_OUTPUT_DIR=$GITHUB_WORKSPACE/novelrt/docs
+
+      - name: Commit Documentation Updates
+        if: ${{ success() }}
+        run: |
+          git add ./docs/*
+          git commit -m "Updated documentation - $GITHUB_SHA"
+          git push origin documentation/$GITHUB_SHA
+
+      - name: Create Pull Request
```

Comment:
nit: If we can trivially acquire https://github.com/cli/cli, that would be better than depending on a third party CI leg
-- It may even be default on the GH Actions machines

That should just be downloading the zip from the release URL and then running `gh auth logic` (which should pick up `with: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}`)
and then doing `gh pr create --title "title" --body "body" --base gh-pages`

I imagine the `create-pull-request@1.0.0` action is doing basically that itself anyways

Reply:
https://github.blog/2021-03-11-scripting-with-github-cli/#using-gh-in-github-actions

It does indeed come built in

Reply:
Finally got around to playing with GH CLI and testing this. Latest commit should include the change iirc